doc/develop/api/api_conventions.rst

```diff
@@ -0,0 +1,58 @@
+.. _api_conventions.rst:
+
+API Conventions
+###############
+
+Zephyr provides a few programming interfaces that users can use for creating their applications.
+Zephyr's API space includes the following areas:
+- C programming interface (function prototypes, structures, and macros), provided in C header files
+- Configuration system (Kconfig)
+- Device driver model (Devicetree)
+
+Other areas, such as the build system (CMake), may also be seen as Zephyr's API. However, any area
+that is not explicitly listed herein is not considered Zephyr's API as of today.
+
+API classification
+******************
+The files that define the Zephyr APIs may contain symbols intended for different usage. The intended
+API usage is designated by the API class. Zephyr APIs are classified as:
+
+- private - these APIs are intended for use within boundary of a software component such as
+            driver or subsystem. Private APIs defined in Zephyr tree are not subject to
```

Comment:
I think we need an actual definition for "software component". AFAICT we intend this to mean the smallest unit we control to compose a system. It seems pretty clear that each shell backend is a different component. Since the scheduler is removed from the kernel in a minimal configuration it might also be one or more components.

I am hesitant to equate component with unit because the "unit_testing" board uses the host toolchain rather than the target toolchain, and it is very reasonable to have a separate unit test suite for each file in a component.